src/ORM/Behavior/Translate/EavEngine.php

```diff
@@ -0,0 +1,431 @@
+            $conditions = [
+                $name . '.model' => $fieldConditions['model'],
+                $name . '.field' => $field,
+            ];
+            foreach ($fieldConditions as $fieldName => $fieldValue) {
+                $conditions[$name . '.' . $fieldName] = $fieldValue;
+            }
+            if (!TableRegistry::exists($name)) {
+                $fieldTable = TableRegistry::get($name, [
+                    'className' => $table,
+                    'alias' => $name,
+                    'table' => $this->_translationTable->table()
+                ]);
+            } else {
+                $fieldTable = TableRegistry::get($name);
+            }
+
+            $this->_table->hasOne($name, [
+                'targetTable' => $fieldTable,
+                'foreignKey' => 'foreign_key',
+                'joinType' => $filter ? 'INNER' : 'LEFT',
+                'conditions' => $conditions,
+                'propertyName' => $field . '_translation'
+            ]);
+        }
+
+        $this->_table->hasMany($targetAlias, [
+            'className' => $table,
+            'foreignKey' => 'foreign_key',
+            'strategy' => $strategy,
+            'conditions' => ["$targetAlias.model" => $fieldConditions['model']],
+            'propertyName' => '_i18n',
+            'dependent' => true
+        ]);
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @param \Cake\Event\Event $event The beforeFind event that was fired.
+     * @param \Cake\ORM\Query $query Query
+     * @param \ArrayObject $options The options for the query
+     * @return void
+     */
+    public function beforeFind(Event $event, Query $query, $options)
+    {
+        $locale = $this->locale();
+
+        if ($locale === $this->config('defaultLocale')) {
+            return;
+        }
+
+        $conditions = function ($field, $locale, $query, $select) {
+            return function ($q) use ($field, $locale, $query, $select) {
+                $q->where([$q->repository()->alias() . '.locale' => $locale]);
+                $alias = $this->_table->alias();
+
+                if ($query->autoFields() ||
+                    in_array($field, $select, true) ||
+                    in_array("$alias.$field", $select, true)
+                ) {
+                    $q->select(['id', 'content']);
+                }
+
+                return $q;
+            };
+        };
+
+        $contain = [];
+        $fields = $this->_config['fields'];
+        $alias = $this->_table->alias();
+        $select = $query->clause('select');
+
+        $changeFilter = isset($options['filterByCurrentLocale']) &&
+            $options['filterByCurrentLocale'] !== $this->_config['onlyTranslated'];
+
+        foreach ($fields as $field) {
+            $name = $alias . '_' . $field . '_translation';
+
+            $contain[$name]['queryBuilder'] = $conditions(
+            $field,
+            $locale,
+            $query,
+            $select
```

Comment:
I'm pretty sure its the code from the original behavior, how would you like to have it look like?